### PR TITLE
feat: dashboard recipe

### PR DESCRIPTION
**Pull request by sattvikc**
## Summary of change

(A few sentences about this PR)

## Related issues

-   Link to issue1 here
-   Link to issue1 here

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates

-   [ ] Changelog has been updated
-   [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `supertokens/constants.go`
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `supertokens/constants.go > version variable`
-   [ ] Had installed and ran the pre-commit hook
-   [ ] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.

### CODE REVIEW

**Review by nkshah2 — APPROVED** [review]

File: recipe/dashboard/api/usersGet.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+			"status":              "OK",
+			"users":               usersResponse.Users,
+			"nextPaginationToken": usersResponse.NextPaginationToken,
+		})
+	}
+
+	var processingGroup sync.WaitGroup
+	processingGroup.Add(len(usersResponse.Users))
+
+	batchSize := 5
```
Comment (rishabhpoddar) [inline recipe/dashboard/api/usersGet.go:67]: discuss the below with me

File: recipe/dashboard/recipeimplementation.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/supertokens/supertokens-golang/recipe/dashboard/dashboardmodels"
+	"github.com/supertokens/supertokens-golang/supertokens"
+)
+
+func makeRecipeImplementation(querier supertokens.Querier, config dashboardmodels.TypeNormalisedInput, appInfo supertokens.NormalisedAppinfo) dashboardmodels.RecipeInterface {
```
Comment (rishabhpoddar) [inline recipe/dashboard/recipeimplementation.go:27]: - querier not needed for this recipe. 
- config not needed to be passed here - it should be passed in the shouldAllowAccess function arg (see node PR)
- appIinfo not needed

File: supertokens/recipeModule.go [hunk 3]
```diff
@@ -18,13 +18,14 @@ package supertokens
...
-	GetAllCORSHeaders     func() []string
-	GetAPIsHandled        func() ([]APIHandled, error)
-	HandleError           func(err error, req *http.Request, res http.ResponseWriter) (bool, error)
-	OnSuperTokensAPIError func(err error, req *http.Request, res http.ResponseWriter)
+	recipeID                   string
+	appInfo                    NormalisedAppinfo
+	HandleAPIRequest           func(ID string, req *http.Request, res http.ResponseWriter, theirHandler http.HandlerFunc, path NormalisedURLPath, method string) error
+	GetAllCORSHeaders          func() []string
+	GetAPIsHandled             func() ([]APIHandled, error)
+	GetAPIIdIfCanHandleRequest func(path NormalisedURLPath, method string) (*string, error)
```
Comment (rishabhpoddar) [inline supertokens/recipeModule.go:26]: You should not change this struct type.. instead in the dashboard recipe, provide an implementation of `ReturnAPIIdIfCanHandleRequest` which overrides the one in the base class (this one)

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: recipe/dashboard/api/dashboard.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+	htmlString, err := (*apiImplementation.DashboardGET)(options, supertokens.MakeDefaultUserContextFromAPI(options.Req))
+	if err != nil {
+		return err
+	}
+
+	options.Res.Header().Set("Content-Type", "text/html; charset=utf-8")
+	options.Res.WriteHeader(200)
+
+	_, err = fmt.Fprint(options.Res, htmlString)
+	return err
```
[the remark this answers is not in the document]
Comment (sattvikc) [reply recipe/dashboard/api/dashboard.go:40]: Done

File: recipe/dashboard/api/implementation.go [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+		bundleDomain := normalizedDomain.GetAsStringDangerous() + normalizedPath.GetAsStringDangerous()
+
+		stInstance, err := supertokens.GetInstanceOrThrowError()
+		if err != nil {
+			return "", err
+		}
+		connectionURI := stInstance.SuperTokens.ConnectionURI
+
+		normalizedDashboardPath, err := supertokens.NewNormalisedURLPath("/dashboard")
```
[the remark this answers is not in the document]
Comment (sattvikc) [reply recipe/dashboard/api/implementation.go:47]: Done

File: recipe/dashboard/api/usersGet.go (hunk 1 above)
Comment (sattvikc) [reply recipe/dashboard/api/usersGet.go:67]: Done

File: recipe/dashboard/recipeimplementation.go (hunk 2 above)
Comment (sattvikc) [reply recipe/dashboard/recipeimplementation.go:27]: Done

File: supertokens/recipeModule.go (hunk 3 above)
Comment (sattvikc) [reply supertokens/recipeModule.go]: Updated as per our discussion.

**Review by rishabhpoddar — APPROVED** [review]

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**